pkg/discovery/monitoring/kubelet/kubelet_monitor.go

```diff
@@ -218,19 +236,28 @@ func parseMetricFamilies(metricFamilies map[string]*dto.MetricFamily) map[string
 				default:
 				}
 			}
-			if name == "" {
-				// This metric is for pod not container
+			if name == "" || name == "POD" {
+				// A metric with name as "" are for pod not container and a metric with name as "POD" is parent cgroup
+				// for this container and tracks stats for all the containers in the pod. Skip these metrics.
 				continue
 			}
 			metricID := fmt.Sprintf("%s/%s/%s", namespace, podName, name)
 			tm, exists := parsed[metricID]
 			if !exists {
 				tm = &throttlingMetric{}
 			}
-			if metricName == kubeclient.ContainerCPUTotal {
-				tm.cpuTotal = metric.Counter.GetValue()
-			} else {
-				tm.cpuThrottled = metric.Counter.GetValue()
+			if metricFamily.GetType() == dto.MetricType_COUNTER {
+				if metricName == kubeclient.ContainerCPUTotal {
+					tm.cpuTotal = metric.Counter.GetValue()
+				} else {
+					tm.cpuThrottled = metric.Counter.GetValue()
+				}
+			} else if metricFamily.GetType() == dto.MetricType_GAUGE {
+				if metricName == kubeclient.ContainerCPUQuota {
+					tm.cpuQuota = metric.Gauge.GetValue()
+				} else {
+					tm.cpuPeriod = metric.Gauge.GetValue()
+				}
```

Comment:
I would support the use of case here. Also I think `if metricFamily.GetType() == dto.MetricType_COUNTER` and `if metricFamily.GetType() == dto.MetricType_GAUGE` are unnecessary, as we have discarded the possibility of any other type at line 217 above.